diff --git a/src/MEFISTO2/aptrte.cxx b/src/MEFISTO2/aptrte.cxx
index 807575a51..c66ffb0f1 100644
--- a/src/MEFISTO2/aptrte.cxx
+++ b/src/MEFISTO2/aptrte.cxx
@@ -40,9 +40,9 @@ extern "C"
       __stdcall
   #endif
   #endif
-      areteideale()//( R3 xyz, R3 direction )
+      areteideale(R &_areteideale)
   {
-    return aretemaxface_;
+      _areteideale = aretemaxface_;
   }
 }
 //calcul de la longueur ideale de l'arete au sommet xyz (z ici inactif)
@@ -244,7 +244,8 @@ void  aptrte( Z   nutysu, R      aretmx,
     ns0 = nudslf[n-1];
     mnpxyd[ns0].x = uvslf[ns0].x;
     mnpxyd[ns0].y = uvslf[ns0].y;
-    mnpxyd[ns0].z = areteideale();//( mnpxyd[ns0], direction );
+    // mnpxyd[ns0].z = areteideale();//( mnpxyd[ns0], direction );
+    mnpxyd[ns0].z = areteideale((R &)mnpxyd[ns0].z);//( mnpxyd[ns0], direction );
 //     MESSAGE("Sommet " << ns0 << ": " << mnpxyd[ns0].x
 //       << " " << mnpxyd[ns0].y << " longueur arete=" << mnpxyd[ns0].z);
 
@@ -299,7 +300,8 @@ void  aptrte( Z   nutysu, R      aretmx,
 //fin   ajout  5/10/2006  ................................................
       mnpxyd[ns].x = uvslf[ns].x;
       mnpxyd[ns].y = uvslf[ns].y;
-      mnpxyd[ns].z = areteideale();//( mnpxyd[ns], direction );
+      // mnpxyd[ns].z = areteideale();//( mnpxyd[ns], direction );
+      mnpxyd[ns1].z = areteideale((R &)mnpxyd[ns1].z);//( mnpxyd[ns1], direction );
 //       MESSAGE("Sommet " << ns << ": " << mnpxyd[ns].x
 //         << " " << mnpxyd[ns].y << " longueur arete=" << mnpxyd[ns].z);
 
@@ -391,7 +393,8 @@ void  aptrte( Z   nutysu, R      aretmx,
     //les 2 coordonnees du point i de sommet nbs
     mnpxyd[ns1].x = uvpti[i].x;
     mnpxyd[ns1].y = uvpti[i].y;
-    mnpxyd[ns1].z = areteideale();//( mnpxyd[ns1], direction );
+    // mnpxyd[ns1].z = areteideale();//( mnpxyd[ns1], direction );
+    mnpxyd[ns1].z = areteideale((R &)mnpxyd[ns1].z);//( mnpxyd[ns1], direction );
     //le numero i du point interne
     mnslig[ns1] = i+1;
     ns1++;
diff --git a/src/MEFISTO2/aptrte.h b/src/MEFISTO2/aptrte.h
index a979f41fa..5a03ec7fe 100644
--- a/src/MEFISTO2/aptrte.h
+++ b/src/MEFISTO2/aptrte.h
@@ -52,6 +52,7 @@
  #else
   #define MEFISTO2D_EXPORT __declspec( dllimport )
  #endif
+ #define F2C_BUILD
 #else
  #define MEFISTO2D_EXPORT
 #endif
