diff --git a/libsrc/occ/occgenmesh.cpp b/libsrc/occ/occgenmesh.cpp
--- a/libsrc/occ/occgenmesh.cpp
+++ b/libsrc/occ/occgenmesh.cpp
@@ -1258,8 +1258,8 @@ namespace netgen
                 for (int k = 1; k <=3; k++)
                   {
                     int n = triangulation->Triangles()(j)(k);
-                    p[k-1] = triangulation->Nodes()(n).Transformed(loc);
-                    par[k-1] = triangulation->UVNodes()(n);
+                    p[k-1] = triangulation->Node(n).Transformed(loc);
+                    par[k-1] = triangulation->UVNode(n);
                   }
 
                 //double maxside = 0;
