diff --git a/src/SMESH/SMESH_Mesh.cxx b/src/SMESH/SMESH_Mesh.cxx
--- a/src/SMESH/SMESH_Mesh.cxx
+++ b/src/SMESH/SMESH_Mesh.cxx
@@ -44,8 +44,10 @@
 #include "DriverDAT_W_SMDS_Mesh.h"
 #include "DriverGMF_Read.hxx"
 #include "DriverGMF_Write.hxx"
+#ifdef WITH_MED
 #include "DriverMED_R_SMESHDS_Mesh.h"
 #include "DriverMED_W_SMESHDS_Mesh.h"
+#endif
 #include "DriverSTL_R_SMDS_Mesh.h"
 #include "DriverSTL_W_SMDS_Mesh.h"
 #include "DriverUNV_R_SMDS_Mesh.h"
@@ -502,6 +504,7 @@ int SMESH_Mesh::UNVToMesh(const char* theFileName)
 
 int SMESH_Mesh::MEDToMesh(const char* theFileName, const char* theMeshName)
 {
+#ifdef WITH_MED
   if ( _isShapeToMesh )
     throw SALOME_Exception(LOCALIZED("a shape to mesh has already been defined"));
   _isShapeToMesh = false;
@@ -536,6 +539,9 @@ int SMESH_Mesh::MEDToMesh(const char* theFileName, const char* theMeshName)
   _myMeshDS->CompactMesh();
 
   return (int) status;
+#else
+  return 0;
+#endif
 }
 
 //=======================================================================
@@ -1434,6 +1440,7 @@ void SMESH_Mesh::ExportMED(const char *        file,
                            bool                theAllElemsToGroup)
 throw(SALOME_Exception)
 {
+#ifdef WITH_MED
   MESSAGE("MED_VERSION:"<< theVersion);
   SMESH_TRY;
 
@@ -1492,6 +1499,7 @@ throw(SALOME_Exception)
   myWriter.Perform();
 
   SMESH_CATCH( SMESH::throwSalomeEx );
+#endif
 }
 
 //================================================================================
@@ -1505,6 +1513,7 @@ void SMESH_Mesh::ExportSAUV(const char *file,
                             bool theAutoGroups)
   throw(SALOME_Exception)
 {
+#ifdef WITH_MED
   std::string medfilename(file);
   medfilename += ".med";
   std::string cmd;
@@ -1538,6 +1547,7 @@ void SMESH_Mesh::ExportSAUV(const char *file,
   cmd += "from medutilities import my_remove ; my_remove(r'" + medfilename + "')";
   cmd += "\"";
   system(cmd.c_str());
+#endif
 }
 
 //================================================================================
diff --git a/src/StdMeshers/StdMeshers_Cartesian_3D.cxx b/src/StdMeshers/StdMeshers_Cartesian_3D.cxx
--- a/src/StdMeshers/StdMeshers_Cartesian_3D.cxx
+++ b/src/StdMeshers/StdMeshers_Cartesian_3D.cxx
@@ -2561,7 +2561,7 @@ namespace
         case 3: // at a corner
         {
           _Node& node = _hexNodes[ subEntity - SMESH_Block::ID_FirstV ];
-          if ( node.Node() > 0 )
+          if ( node.Node() != nullptr )
           {
             if ( node._intPoint )
               node._intPoint->Add( _eIntPoints[ iP ]->_faceIDs, _eIntPoints[ iP ]->_node );

 //================================================================================
diff --git a/src/SMDS/SMDS_ElementFactory.hxx b/src/SMDS/SMDS_ElementFactory.hxx
--- a/src/SMDS/SMDS_ElementFactory.hxx
+++ b/src/SMDS/SMDS_ElementFactory.hxx
@@ -382,7 +382,7 @@ class SMDS_ElementChunk
 public:
 
   SMDS_ElementChunk( SMDS_ElementFactory* factory = 0, int id0 = 0 );
-  ~SMDS_ElementChunk();
+  virtual ~SMDS_ElementChunk();
 
   //! Return an element by an index [0,ChunkSize()]
   SMDS_MeshElement* Element(int index) { return & myElements[index]; }
