diff --git a/src/SMESH/SMESH_Mesh.cxx b/src/SMESH/SMESH_Mesh.cxx
--- a/src/SMESH/SMESH_Mesh.cxx
+++ b/src/SMESH/SMESH_Mesh.cxx
@@ -42,12 +42,14 @@
 #include "utilities.h"
 
 #include "DriverDAT_W_SMDS_Mesh.h"
 #include "DriverGMF_Read.hxx"
 #include "DriverGMF_Write.hxx"
+#ifdef WITH_MED
 #include "DriverMED_R_SMESHDS_Mesh.h"
 #include "DriverMED_W_SMESHDS_Mesh.h"
+#endif
 #include "DriverSTL_R_SMDS_Mesh.h"
 #include "DriverSTL_W_SMDS_Mesh.h"
 #include "DriverUNV_R_SMDS_Mesh.h"
 #include "DriverUNV_W_SMDS_Mesh.h"
 #ifdef WITH_CGNS
@@ -500,10 +502,11 @@ int SMESH_Mesh::UNVToMesh(const char* theFileName)
 //purpose  :
 //=======================================================================
 
 int SMESH_Mesh::MEDToMesh(const char* theFileName, const char* theMeshName)
 {
+#ifdef WITH_MED
   if ( _isShapeToMesh )
     throw SALOME_Exception(LOCALIZED("a shape to mesh has already been defined"));
   _isShapeToMesh = false;
 
   DriverMED_R_SMESHDS_Mesh myReader;
@@ -534,10 +537,13 @@ int SMESH_Mesh::MEDToMesh(const char* theFileName, const char* theMeshName)
 
   _myMeshDS->Modified();
   _myMeshDS->CompactMesh();
 
   return (int) status;
+#else
+  return 0;
+#endif
 }
 
 //=======================================================================
 //function : STLToMesh
 //purpose  :
@@ -1425,10 +1431,11 @@ void SMESH_Mesh::ExportMED(const char *        file,
                            bool                theAutoDimension,
                            bool                theAddODOnVertices,
                            double              theZTolerance,
                            bool                theAllElemsToGroup)
 {
+#ifdef WITH_MED
   MESSAGE("MED_VERSION:"<< theVersion);
 
   Driver_Mesh::Status status;
   SMESH_TRY;
 
@@ -1488,10 +1495,11 @@ void SMESH_Mesh::ExportMED(const char *        file,
 
   SMESH_CATCH( SMESH::throwSalomeEx );
 
   if ( status == Driver_Mesh::DRS_TOO_LARGE_MESH )
     throw TooLargeForExport("MED");
+#endif
 }
 
 //================================================================================
 /*!
  * \brief Export the mesh to a SAUV file
@@ -1500,10 +1508,11 @@ void SMESH_Mesh::ExportMED(const char *        file,
 
 void SMESH_Mesh::ExportSAUV(const char *file, 
                             const char* theMeshName, 
                             bool theAutoGroups)
 {
+#ifdef WITH_MED
   std::string medfilename(file);
   medfilename += ".med";
   std::string cmd;
 #ifdef WIN32
   cmd = "%PYTHONBIN% ";
@@ -1539,10 +1548,11 @@ void SMESH_Mesh::ExportSAUV(const char *file,
 #endif
   cmd += "-c \"";
   cmd += "from medutilities import my_remove ; my_remove(r'" + medfilename + "')";
   cmd += "\"";
   system(cmd.c_str());
+#endif
 }
 
 //================================================================================
 /*!
  * \brief Export the mesh to a DAT file
